tap_github/repository_streams.py

```diff
@@ -1348,6 +1348,11 @@ class PullRequestCommits(GitHubRestStream):
         ),
     ).to_dict()
 
+    def post_process(self, row: dict, context: Optional[Dict[str, str]] = None) -> dict:
```

Comment:
@scottmtp could you try adding `pull_number` to `state_partitioning_keys` instead and see if that fixes it?
Also, could you add a small test for this change?

Thanks for your PR! :)

Reply:
If it doesn't work, make sure to call the parent `post_process` function to ensure that you are also passing the `repo_id`

Reply:
Yes, adding `pull_number` to `state_partitioning_keys` does indeed fix the issue.

Reply:
This works, but beware of the implication: when you add `pull_number` to `state_partitioning_keys`, you're also expanding the state file to keep track of the state of each individual pull request in the stream, which might lead to a gigantic number of state bookmarks for bigger repos. Considering that PRs are "short lived" for most of them, I'm not sure it's worth keeping so much state overhead.

Reply:
@ericboucher yeah, Laurent is correct and it would also be a breaking change by making old state incompatible. `post_process` seems like the right approach 👍